pkgs/top-level/perl-packages.nix

```diff
@@ -1521,6 +1521,27 @@ let
     };
   };
 
+  BioPerl = buildPerlPackage {
+    pname = "BioPerl";
+    version = "1.7.8";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/C/CJ/CJFIELDS/BioPerl-1.7.8.tar.gz";
+      sha256 = "c490a3be7715ea6e4305efd9710e5edab82dabc55fd786b6505b550a30d71738";
+    };
+    buildInputs = [ TestMemoryCycle TestWeaken ];
+    # We add the dependencies manually
+    propagatedBuildInputs = [DataStag ] ++
+                            (with perlPackages; [Error Graph HTTPMessage IOString IOStringy IPCRun LWP ListMoreUtils SetScalar TestMost TestRequiresInternet URI XMLDOM XMLLibXML XMLSAX XMLSAXBase XMLSAXWriter XMLTwig XMLWriter YAML
+                                                ])
+                            ++ (with perlPackages; [DBFile ModuleBuild TestDeep TestWarn TestException TestDifferences])
+                            ++ [perlPackages.libxml_perl];
```

Comment:
These need listing as one array, e.g.

```
propagatedBuildInputs = [ DataStag Error Graph HTTPMessage ... DBFile ... ];
```

also, `Test::*` module dependencies here should go into `buildInputs`.